src/components/screens/Authenticator/Authenticator.container.vue

```diff
@@ -127,25 +133,33 @@ export default {
           this.replaceRoute('EmailCode');
           break;
 
-        case name === 'SignUp' && isSmsOtp:
-        case name === 'RegularPassword' && isSmsOtp:
+        case name === 'SignIn' && isSmsOtp && !this.isDeviceRemembered:
```

Comment:
what do you mean?

Reply:
A lot of conditions. It not relates for current PR, but I guess - it is potential problem

Reply:
agreed with @lamartire 

maybe use specific component for determine required route? because this is very big piece of logic which looks very difficult to understand and read

Reply:
Will be refactored later. I agree that it's a big ball